networkx/generators/joint_degree_seq.py

```diff
@@ -0,0 +1,286 @@
+#    Copyright (C) 2016 by
+#    Minas Gjoka
+#    All rights reserved.
+#    BSD license.
+#
+# Author:  Minas Gjoka (minas.gjoka@gmail.com)
+"""Generate graphs with a given joint degree """
+
+import random
+
+import networkx as nx
+
+__all__ = ['is_valid_joint_degree',
+           'joint_degree_model']
+
+
+def is_valid_joint_degree(nkk):
+    """ Checks whether the given joint degree dictionary (nkk) is realizable 
+        as a simple graph by evaluating five necessary and sufficient conditions.
+        
+    Here is the list of five conditions that an input nkk needs to satisfy for
+    simple graph realizability:
+    - Condition    1:  nkk[k][l]  is integer for all k,l    
+    - Condition    2:  sum(nkk[k])/k = number of nodes with degree k, is an integer    
+    - Conditions 3,4: number of edges between k and l cannot exceed maximum possible number of edges 
+    - Condition    5: nkk[k][k] is an even integer i.e. stubs are counted twice for equal degree pairs.
+                      this is an assumption that the method joint_degree_model expects to be true.
+           
+
+    Parameters
+    ----------
+    nkk :  dictionary of dictionary of integers
+        joint degree dictionary. for nodes of degree k (first level of dict) and
+        nodes of degree l (second level of dict) describes the number of edges        
+    
+    
+    Returns
+    -------
+    boolean
+        returns true if given joint degree dictionary is realizable, else returns false.
+
+    
+    References
+    ----------
+    [1] M. Gjoka, M. Kurant, A. Markopoulou, "2.5K Graphs: from Sampling to Generation",
+    IEEE Infocom, 2013.
+    
+    """
+    
+            
+    nk = {}
+    for k in nkk:
+        if k>0:
+            k_size = float( sum(val for val in nkk[k].itervalues()) ) / float( k )
```

Comment:
This would be simpler as follows. First, add `from __future__ import division` to the head of the file (after the module docstring), then do:

    k_size = sum(nkk[k].values()) / k